### ISSUE TITLE
wrong recommendation for default_trait_access

### ISSUE THREAD

**Issue by tspiteri** [rust-lang/rust-clippy#2879]
The lint gives a wrong recommendation for the code below, suggesting that `<Foo as Default>::default()` be replaced with `Foo::default()`. It is true that it is a bad idea to have an inherent `default` method with behaviour that is unlike the `Default` implementation, but the recommendation is still wrong.

```rust
#[derive(Default)]
struct Foo(i32);

impl Foo {
    fn default() -> Foo { Foo(1) }
}

fn main() {
    let should_be_one = Foo::default();
    println!("{}", should_be_one.0);
    let should_be_zero = <Foo as Default>::default();
    println!("{}", should_be_zero.0);
}
```

Maybe there could be a lint to check that there is no inherent method called `default`, which would mitigate this, because then at least if all warnings are removed the behaviour is still good. It still wouldn't fix this if the questionable inherent method is in another crate.

Another thing: for this particular example the warning is I think a false positive. `<Foo as Default>::default()` should *not* be flagged by this lint, as the type name is there anyway.
